torch/csrc/api/src/serialize/default.cpp

```diff
@@ -0,0 +1,67 @@
+#include <torch/serialize/default.h>
+
+#include <torch/serialize/base.h>
+#include <torch/tensor.h>
+#include <torch/utils.h>
+
+#include <torch/csrc/jit/export.h>
+#include <torch/csrc/jit/import.h>
+#include <torch/csrc/jit/script/module.h>
+
+#include <memory>
+#include <string>
+
+namespace torch {
+namespace serialize {
+
+namespace detail {
+inline std::string replace_dots_with_underscores(std::string string) {
+  for (char& c : string) {
+    if (c == '.') {
+      c = '_';
+    }
+  }
+  return string;
+}
+} // namespace detail
+
+DefaultWriter::DefaultWriter(std::string filename) : filename_(filename) {}
+
+void DefaultWriter::write(
+    const std::string& key,
+    const Tensor& tensor,
+    bool is_buffer) {
+  auto patched_key = detail::replace_dots_with_underscores(key);
+  module_.register_parameter(patched_key, tensor, is_buffer);
```

Comment:
so it represents the entire module as a gigantic flat module even in presence of submodules?